torch/ao/quantization/observer.py

```diff
@@ -1131,66 +1131,74 @@ def _non_linear_param_search(self) -> Tuple[torch.Tensor, torch.Tensor]:
         new_max = self.min_val + bin_width * (end_bin + 1)
         return new_min, new_max
 
-    def _adjust_min_max(
-        self, combined_min: torch.Tensor, combined_max: torch.Tensor, upsample_rate: int
-    ) -> Tuple[torch.Tensor, torch.Tensor, int, int]:
-        # We ensure that:
-        # (combined_max - combined_min)/(downsample_rate*Nbins) = (max - min)/(upsample_rate*Nbins)
-        # This allows us to have a common grid of resolution s, where we can align
-        # the input histogram
-        # start_idx maps min_val to the histogram bin index.
-
-        # Compute the width of histogram bins is a straightforward solution, where
-        # hist_bin_width = (self.max_val - self.min_val) / (self.bins * upsample_rate)
-        # Underflow happens if the numerator is close to the smallest positive subnormal number of FP32
-        # Therefore, we avoid such division operation.
-        downsample_rate = int(
-            torch.ceil(
-                ((combined_max - combined_min) / (self.max_val - self.min_val)) * upsample_rate
-            ).item()
+    @staticmethod
+    def upscale_histogram(
+        histogram: torch.Tensor,
+        orig_min: torch.Tensor,
+        orig_max: torch.Tensor,
+        update_min: torch.Tensor,
+        update_max: torch.Tensor,
+        n_bins: int,
+        upsample_rate: int = 4,
```

Comment:
why is this not the same with L967?

Reply:
Good point. The original code was written in this way, essentially like a static_method. 
Do you think it's better if I rewrote everything to use just self. everywhere? Or do you like the more explicit static_method kind of setup? 

I removed the default from the static_method - but calling with self.upsample_rate in _combine_histograms. If you want me to go full ham, I can remove most passing and call e.g. self. histogram everywhere instead of passing it.

Reply:
yeah it seems that we don't really need to make this a staticmethod, I feel we can just do `self` everywhere and just remove the args

Reply:
Fixed this and made things non-static